### ISSUE TITLE
Join Meeting Password: Is there any way to skip the password prompt for join meeting?

### ISSUE THREAD

**Issue by employee-proctur** [chaimPaneth/react-native-zoom-bridge#7]
As I can see, `ZoomBridge.joinMeeting` needs two parameters to join a meeting for a user. Later on in this process, a dialog box appears asking for a meeting password. Is there any way to pass the password as well in this method instead of prompting user to enter the password.

```javascript
async joinMeeting(){
  try{
      let result = await ZoomBridge.joinMeeting("John Doe", '77794198988');
      console.log('Zoom Join Meeting Response: ', result);
  }
  catch(e)
  {
      console.log('ZOOM JOIN Error: ', e);
      console.warn({e});
   }
 }
```

In our use case, we are showing a list of live classes to many students (might not be registered users on Zoom) in our app. Now they are already logged into our app and for smooth functioning, students expect to join the live class with just one click.

So, is there any way to achieve this?

**Comment by KawaljeetSBagga** [comment]
Hey seems like we can only join meeting with display name and meeting id as of now, and if there's a password required, then the user has to enter it manually. I have added a method on both the platforms iOS and android inside your package which are as follows:
**iOS**:

>RCT_EXPORT_METHOD(
joinMeetingWithPassword: (NSString *)displayName
withMeetingNo: (NSString *)meetingNo
withPassword: (NSString *)password
withResolve: (RCTPromiseResolveBlock)resolve
withReject: (RCTPromiseRejectBlock)reject
)
{
@try {
meetingPromiseResolve = resolve;
meetingPromiseReject = reject;
MobileRTCMeetingService *ms = [[MobileRTC sharedRTC] getMeetingService];
if (ms) {
ms.delegate = self;
NSDictionary *paramDict = @{
kMeetingParam_Username: displayName,
kMeetingParam_MeetingNumber: meetingNo,
kMeetingParam_MeetingPassword: password
};
MobileRTCMeetError joinMeetingResult = [ms joinMeetingWithDictionary:paramDict];
NSLog(@"joinMeeting, joinMeetingResult=%d", joinMeetingResult);
}
} @catch (NSError *ex) {
reject(@"ERR_UNEXPECTED_EXCEPTION", @"Executing joinMeeting", ex);
}
}

**Android**:

> @ReactMethod
public void joinMeetingWithPassword(
final String displayName,
final String meetingNo,
final String password,
Promise promise
) {
try {
meetingPromise = promise;
ZoomSDK zoomSDK = ZoomSDK.getInstance();
if(!zoomSDK.isInitialized()) {
promise.reject("ERR_ZOOM_JOIN", "ZoomSDK has not been initialized successfully");
return;
}
final MeetingService meetingService = zoomSDK.getMeetingService();
JoinMeetingOptions opts = new JoinMeetingOptions();
JoinMeetingParams params = new JoinMeetingParams();
params.displayName = displayName;
params.meetingNo = meetingNo;
params.password = password;
int joinMeetingResult = meetingService.joinMeetingWithParams(reactContext.getCurrentActivity(), params, opts);
Log.i(TAG, "joinMeeting, joinMeetingResult=" + joinMeetingResult);
if (joinMeetingResult != MeetingError.MEETING_ERROR_SUCCESS) {
promise.reject("ERR_ZOOM_JOIN", "joinMeeting, errorCode=" + joinMeetingResult);
}
} catch (Exception ex) {
promise.reject("ERR_UNEXPECTED_EXCEPTION", ex);
}
}

I have checked that "joinMeetingWithPassword" methods on iOS simulator iPhone 11 and Android physical device One Plus 7T works well, so I request you to go ahead and update the changes in the package with iOS and android methods. LMK if I need to do a PR.
